zerver/lib/actions.py

```diff
@@ -270,6 +270,27 @@ def get_topic_history_for_stream(user_profile: UserProfile,
 
     return generate_topic_history_from_db_rows(rows)
 
+def get_topic_history_for_web_public_stream(recipient: Recipient) -> List[Dict[str, Any]]:
+    cursor = connection.cursor()
+    query = '''
+    SELECT
+        "zerver_message"."subject" as topic,
```

Comment:
I think we need to handle the fact that two topics might share the same canonical name?

Reply:
Unless I'm mistaken, that is handled by `generate_topic_history_from_db_rows` and the `ORDER by max("message_message".id) DESC`, right?

Reply:
oh, ok, yeah, I guess we're fine here